code/frameworks/angular/src/client/angular-beta/utils/PropertyExtractor.ts

```diff
@@ -13,53 +15,108 @@ const uniqueArray = (arr: any[]) => {
   return arr.flat(Number.MAX_VALUE).filter((value, index, self) => self.indexOf(value) === index);
 };
 
-/** Restricted Imports */
-const RESTRICTED_IMPORTS = [
+const analyzeRestricted = (ngModule: NgModule) => {
   /**
-   * BrowserAnimationsModule imports BrowserModule, which is restricted,
+   * BrowserModule is restricted,
    * because bootstrapApplication API, which mounts the component to the DOM,
    * automatically imports BrowserModule
    */
-  BrowserAnimationsModule,
+  if (ngModule === BrowserModule) {
+    return [true];
+  }
   /**
-   * NoopAnimationsModule imports BrowserModule, which is restricted,
+   * BrowserAnimationsModule imports BrowserModule, which is restricted,
    * because bootstrapApplication API, which mounts the component to the DOM,
    * automatically imports BrowserModule
    */
-  NoopAnimationsModule,
+  if (ngModule === BrowserAnimationsModule) {
+    return [true, provideAnimations()];
+  }
   /**
-   * BrowserModule is restricted,
+   * NoopAnimationsModule imports BrowserModule, which is restricted,
    * because bootstrapApplication API, which mounts the component to the DOM,
    * automatically imports BrowserModule
    */
-  BrowserModule,
-];
+  if (ngModule === NoopAnimationsModule) {
+    return [true, provideNoopAnimations()];
+  }
+
+  return [false];
+};
+
+const REMOVED_MODULES = new InjectionToken('REMOVED_MODULES');
 
 /**
- * Extract Imports from NgModule
- *
- * CommonModule is always imported
+ * Analyze NgModule Metadata
  *
- * metadata.imports are flattened deeply and extracted into a new array
+ * - Removes Restricted Imports
+ * - Extracts providers from ModuleWithProviders
+ * - Flattens imports
+ * - Returns a new NgModuleMetadata object
  *
- * - If ModuleWithProviders (e.g. forRoot() & forChild() ) is used, the ngModule is extracted without providers.
  *
  */
-export const extractImports = (metadata: NgModuleMetadata) => {
-  const imports = [CommonModule];
+const analyzeMetadata = (metadata: NgModuleMetadata) => {
```

Comment:
Can we add some tests for `analyzeMetadata`? Because quite a lot is happening here.